kvbc/src/categorization/kv_blockchain.cpp

```diff
@@ -598,4 +603,19 @@ void KeyValueBlockchain::writeSTLinkTransaction(const BlockId block_id, RawBlock
   block_chain_.setAddedBlockId(new_block_id);
 }
 
+std::string KeyValueBlockchain::getPruningStatus() {
+  std::ostringstream oss;
+  std::unordered_map<std::string, std::string> result;
+
+  result.insert(toPair(getName(versioned_num_of_deletes_keys_), versioned_num_of_deletes_keys_.Get().Get()));
+  result.insert(toPair(getName(immutable_num_of_deleted_keys_), immutable_num_of_deleted_keys_.Get().Get()));
+  result.insert(toPair(getName(merkle_num_of_deleted_keys_), merkle_num_of_deleted_keys_.Get().Get()));
```

Comment:
This is not thread safe. These counters are not atomic, but `getPruningStatus` is called from the diagnostic handler thread. You can either make the variables AtomicCounters  (which will slow down updates which are much more frequent than status access) or use the [aggregator api](https://github.com/vmware/concord-bft/blob/master/util/src/Metrics.cpp#L130) to access the metrics in a thread safe manner. This has no cost to updates, but it will hold a lock that may cause contention if another thread is trying to update its metrics. Either way is probably ok for this specific case.

Reply:
Oh, missed that, thanks. I updated to using the Aggregator API to access the counters in a thread safe manner